src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -1116,43 +1117,30 @@ private bool CheckMembersDistance(DatabaseObservationState state, out string rea
             return true;
         }
 
-        private bool ShouldGiveMoreTimeBeforeMovingToRehab(DateTime lastSuccessfulUpdate, TimeSpan? databaseUpTime)
-        {
-            if (databaseUpTime.HasValue)
-            {
-                if (databaseUpTime.Value.TotalMilliseconds < _moveToRehabTimeMs)
-                {
-                    return true;
-                }
-            }
+        private bool ShouldGiveMoreTimeBeforeMovingToRehab(DateTime lastSuccessfulUpdate, TimeSpan? databaseUpTime) => 
+            ShouldGiveMoreGrace(lastSuccessfulUpdate, databaseUpTime, _moveToRehabTimeMs);
 
-            return ShouldGiveMoreGrace(lastSuccessfulUpdate, databaseUpTime, _moveToRehabTimeMs);
-        }
+        private bool ShouldGiveMoreTimeBeforeRotating(DateTime lastSuccessfulUpdate, TimeSpan? databaseUpTime) => 
+            ShouldGiveMoreGrace(lastSuccessfulUpdate, databaseUpTime, _rotateGraceTimeMs);
 
-        private bool ShouldGiveMoreTimeBeforeRotating(DateTime lastSuccessfulUpdate, TimeSpan? databaseUpTime)
+        private bool ShouldGiveMoreGrace(DateTime lastSuccessfulUpdate, TimeSpan? databaseUpTime, long graceMs)
         {
             if (databaseUpTime.HasValue)
             {
-                if (databaseUpTime.Value.TotalMilliseconds > _rotateGraceTimeMs)
+                if (databaseUpTime.Value.TotalMilliseconds > graceMs)
```

Comment:
isn't that covered by next case? also it returns false here so basically if 'database is up for more than grace' then no grace will be given, then for example lastSuccesfulUpdate is not taken into account? I'm wondering if this should not be entirely removed?